giraffe/src/transforms/band.ts

```diff
@@ -0,0 +1,210 @@
+    : []
+
+  const bands = Array.isArray(mappings)
+    ? mappings.reduce((bandLines, map, index) => {
+        const lineName = columnKeysWithoutResult.reduce((accum, current) => {
+          return `${accum}${mappings[index][current]}`
+        }, '')
+        if (!minMaxMap[lineName]) {
+          minMaxMap[lineName] = {}
+        }
+        if (map[RESULT] === MAX) {
+          if (!minMaxMap[lineName][MAX]) {
+            minMaxMap[lineName][MAX] = lineData[index]
+          }
+        } else if (map[RESULT] === MIN) {
+          if (!minMaxMap[lineName][MIN]) {
+            minMaxMap[lineName][MIN] = lineData[index]
+          }
+        } else {
+          bandLines.push({
+            ...lineData[index],
+            lineName,
+            fill: bandFillColors[bandLines.length],
+          })
+        }
+        return bandLines
+      }, [])
+    : []
+
+  bands.forEach(line => {
+    if (minMaxMap[line.lineName][MIN]) {
+      line[MIN] = {...minMaxMap[line.lineName][MIN], fill: line.fill}
+    }
+    if (minMaxMap[line.lineName][MAX]) {
+      line[MAX] = {...minMaxMap[line.lineName][MAX], fill: line.fill}
+    }
+  })
+  return bands as Band[]
+}
+
+export const getBandIndexMap = (
+  fillColumnMap: ColumnGroupMap
+): BandIndexMap => {
+  const bandIndices = {
+    rowIndices: [],
+    minIndices: [],
+    maxIndices: [],
+  }
+
+  if (Array.isArray(fillColumnMap.mappings)) {
+    fillColumnMap.mappings.forEach((line, index) => {
+      if (line[RESULT] === MAX) {
+        bandIndices.maxIndices.push(index)
+      } else if (line[RESULT] === MIN) {
+        bandIndices.minIndices.push(index)
+      } else {
+        bandIndices.rowIndices.push(index)
+      }
+    })
+  }
+
+  return bandIndices
+}
+
+export const getBandLineIndices = (fillColumnMap: ColumnGroupMap): number[] => {
+  const bandLineIndices = Array.isArray(fillColumnMap.mappings)
+    ? fillColumnMap.mappings.reduce((indices, line, index) => {
+        if (line[RESULT] && line[RESULT] !== MAX && line[RESULT] !== MIN) {
+          indices.push(index)
+        }
+        return indices
+      }, [])
+    : []
+
+  return bandLineIndices as number[]
+}
+
+export const getBandName = (fillColumnMap: ColumnGroupMap): string => {
+  return Array.isArray(fillColumnMap.mappings)
+    ? fillColumnMap.mappings.reduce((name, line) => {
+        if (!name && line[RESULT] !== MAX && line[RESULT] !== MIN) {
+          name = line[RESULT]
+        }
+        return name
+      }, '')
+    : ''
+}
+
+export const mapCumulativeValuesToTimeRange = (
+  timesCol: NumericColumnData,
+  valuesCol: NumericColumnData,
+  fillCol: NumericColumnData
+): CumulativeValuesByTime => {
+  const cumulativeValues: CumulativeValuesByTime = {}
+
+  timesCol.forEach((time, index) => {
+    if (!cumulativeValues[time]) {
+      cumulativeValues[time] = {}
+    }
+    const maxGroup = fillCol[index]
+
+    if (!cumulativeValues[time][maxGroup - 1]) {
+      for (let i = 0; i < maxGroup; i += 1) {
```

Comment:
was curious about `i += 1` vs `i++`. i've noticed them used interchangeably in giraffe and even this PR. is there a preference or reason to use one or the other?

Reply:
In this usage, they are the same. I generally avoid ++ unless there is a specific use case that i need.